src/visualizations/config/adapters/dhis_highcharts/index.js

```diff
@@ -135,6 +137,25 @@ export default function({ store, layout, el, extraConfig, extraOptions }) {
         config.series = addTrendLines(_layout, config.series, stacked)
     }
 
+    // DHIS2-147 add legendset to Column and Bar
+    /* 
+    ** Note: This needs to go last, after all other data manipulation is done, as it changes 
+    ** the format of the data prop from an array of values to an array of objects with y and color props.
+    */
+    const legendSets = extraOptions.legendSets
+
+    if (legendSets?.length && isLegendSetType(layout.type)) {
+        if (_layout.legendDisplayStrategy === LEGEND_DISPLAY_STRATEGY_BY_DATA_ITEM) {
+            config.series = config.series.map(seriesObj => {
```

Comment:
Shouldn't this filter out the dual category fake serie?
I guess `applyLegendSet` is not changing the serie's data if values are `null` anyway...

Reply:
https://github.com/dhis2/analytics/blob/6f73d83ba4f07464d2e356cfef7bc298ed8ca175/src/visualizations/config/adapters/dhis_highcharts/legendSet.js#L11 filters out both 2C fake series and Column series with custom type `line` 👍

Reply:
Right, I overlooked that one and now I remember we discussed it already 👍